src/sentry/integrations/pipeline.py

```diff
@@ -78,17 +80,30 @@ def _finish_pipeline(self, data):
             if created:
                 idp.update(config=identity_config)
 
-            Identity.objects.get_or_create(
-                idp=idp,
-                user=self.request.user,
-                external_id=identity['external_id'],
-                defaults={
-                    'status': IdentityStatus.VALID,
-                    'scopes': identity['scopes'],
-                    'data': identity['data'],
-                    'date_verified': timezone.now(),
-                },
-            )
+            identity_data = {
+                'status': IdentityStatus.VALID,
+                'scopes': identity['scopes'],
+                'data': identity['data'],
+                'date_verified': timezone.now(),
+            }
+
+            try:
+                Identity.objects.get_or_create(
```

Comment:
if it exists do you want to update the defaults?

Reply:
Yeah, it's a change of the identity, so yes.

Reply:
Right, I mean do you want to add the code to do that?

Reply:
Oh I see hat you're saying. I'll open a 2nd PR for that too.